mysql-persistence/src/main/java/com/netflix/conductor/mysql/MySQLDataSourceProvider.java

```diff
@@ -0,0 +1,54 @@
+package com.netflix.conductor.mysql;
+
+import com.zaxxer.hikari.HikariDataSource;
+
+import org.flywaydb.core.Flyway;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.sql.DataSource;
+
+public class MySQLDataSourceProvider implements Provider<DataSource> {
```

Comment:
@Singleton

Reply:
I don't think that marking the Provider as a singleton is helpful in this case, and I also don't think that having it provide a singleton is what we want.

If the concern is that the application uses a singleton, we should handle that in the binding configuration, not the provider.  The module does in fact bind this a singleton.  Is there a different behavior that you would like to see?

Reply:
https://github.com/Netflix/conductor/pull/591/files#diff-a46b98ee0f51900430a24ce12d0ccc8fR23

Reply:
we are fine here since data source is bound as singleton in guice module.